src/Data/PQueue/Prio/Max.hs

```diff
@@ -116,394 +112,5 @@ module Data.PQueue.Prio.Max (
   )
   where
 
-import Control.Applicative (liftA2)
-import Data.Maybe (fromMaybe)
 import Data.PQueue.Prio.Max.Internals
-
-#if MIN_VERSION_base(4,9,0)
-import Data.Semigroup (Semigroup((<>)))
-#endif
-
-import Prelude hiding (map, filter, break, span, takeWhile, dropWhile, splitAt, take, drop, (!!), null)
-import qualified Data.Foldable as F
-
-import qualified Data.PQueue.Prio.Min as Q
-
-#ifdef __GLASGOW_HASKELL__
-import Text.Read (Lexeme(Ident), lexP, parens, prec,
-  readPrec, readListPrec, readListPrecDefault)
-#else
-build :: ((a -> [a] -> [a]) -> [a] -> [a]) -> [a]
-build f = f (:) []
-#endif
-
-first' :: (a -> b) -> (a, c) -> (b, c)
-first' f (a, c) = (f a, c)
-
-#if MIN_VERSION_base(4,9,0)
-instance Ord k => Semigroup (MaxPQueue k a) where
-  (<>) = union
-#endif
-
-instance Ord k => Monoid (MaxPQueue k a) where
-  mempty = empty
-#if !MIN_VERSION_base(4,11,0)
-  mappend = union
-#endif
-  mconcat = unions
-
-instance (Ord k, Show k, Show a) => Show (MaxPQueue k a) where
-  showsPrec p xs = showParen (p > 10) $
-    showString "fromDescList " . shows (toDescList xs)
-
-instance (Read k, Read a) => Read (MaxPQueue k a) where
-#ifdef __GLASGOW_HASKELL__
-  readPrec = parens $ prec 10 $ do
-    Ident "fromDescList" <- lexP
-    xs <- readPrec
-    return (fromDescList xs)
-
-  readListPrec = readListPrecDefault
-#else
-  readsPrec p = readParen (p > 10) $ \r -> do
-    ("fromDescList",s) <- lex r
-    (xs,t) <- reads s
-    return (fromDescList xs,t)
-#endif
-
-instance Functor (MaxPQueue k) where
-  fmap f (MaxPQ q) = MaxPQ (fmap f q)
-
-instance Ord k => Foldable (MaxPQueue k) where
-  foldr f z (MaxPQ q) = foldr f z q
-  foldl f z (MaxPQ q) = foldl f z q
-
-  length = size
-  null = null
-
--- | Traverses in descending order. 'mapM' is strictly accumulating like
--- 'mapMWithKey'.
-instance Ord k => Traversable (MaxPQueue k) where
-  traverse f (MaxPQ q) = MaxPQ <$> traverse f q
-  mapM = mapMWithKey . const
-  sequence = mapM id
-
--- | /O(1)/. Returns the empty priority queue.
-empty :: MaxPQueue k a
-empty = MaxPQ Q.empty
-
--- | /O(1)/. Constructs a singleton priority queue.
-singleton :: k -> a -> MaxPQueue k a
-singleton k a = MaxPQ (Q.singleton (Down k) a)
-
--- | Amortized /O(1)/, worst-case /O(log n)/. Inserts
--- an element with the specified key into the queue.
-insert :: Ord k => k -> a -> MaxPQueue k a -> MaxPQueue k a
-insert k a (MaxPQ q) = MaxPQ (Q.insert (Down k) a q)
```

Comment:
Why did you move all the functions to `Data.PQueue.Prio.Max` instead of only the instances (and fucntions needed for the instances)?

Reply:
It was much, much easier.